### PR TITLE
feature/pt-5/authentication

### PR DESCRIPTION
**Pull request by dungeon-slave**
first PR of 5th task, video will be in next

### CODE REVIEW

File: core_ui/lib/src/widgets/app_button.dart [hunk 1]
```diff
@@ -15,7 +15,8 @@ class AppButton extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return ElevatedButton(
-      style: ElevatedButton.styleFrom(backgroundColor: Theme.of(context).indicatorColor),
+      style: ElevatedButton.styleFrom(
+          backgroundColor: Theme.of(context).indicatorColor),
```
Comment (ibondarovich) [inline core_ui/lib/src/widgets/app_button.dart:19]: ,

File: data/lib/providers/remote/firebase_auth_provider.dart [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+  Future<String> emailSignUp({
+    required EmailSignUpModel data,
+  }) async {
+    return (await _firebaseAuth.createUserWithEmailAndPassword(
+      email: data.email,
+      password: data.password,
+    ))
+        .user!
+        .uid; //TODO remove ! operator
```
Comment (ibondarovich) [inline data/lib/providers/remote/firebase_auth_provider.dart:22]: ?

File: domain/lib/usecase/authentication/save_user_usecase.dart [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import 'package:domain/repositories/authentication_repository.dart';
+import 'package:domain/usecase/usecase.dart';
+
+class SaveUserUseCase implements AsyncUseCase<String, void> {
+  final AuthenticationRepository _authenticationRepository;
+
+  const SaveUserUseCase(
+      {required AuthenticationRepository authenticationRepository})
```
Comment (ibondarovich) [inline domain/lib/usecase/authentication/save_user_usecase.dart:8]: formatting

File: domain/lib/usecase/text_scale/set_text_scale_usecase.dart [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+class SetTextScaleUseCase implements AsyncUseCase<double, void> {
+  final TextScaleRepository _textScaleRepository;
+
+  const SetTextScaleUseCase({
+    required TextScaleRepository textScaleRepository,
+  }) : _textScaleRepository = textScaleRepository;
+
+  @override
+  Future<void> execute(double input) async => await _textScaleRepository.saveTextScale(input);
```
Comment (ibondarovich) [inline domain/lib/usecase/text_scale/set_text_scale_usecase.dart:12]: ,

File: features/dishes_menu/lib/src/bloc/dishes_menu_screen/dishes_menu_bloc.dart [hunk 5]
```diff
@@ -28,11 +31,11 @@ class MenuBloc extends Bloc<MenuEvent, MenuState> {
     add(InitEvent());
   }
 
-  void _init(InitEvent event, Emitter<MenuState> emit) async {
-    emit(state.copyWith(isLoading: true));
+  Future<void> _init(InitEvent event, Emitter<MenuState> emit) async {
     try {
       final List<DishTypeModel> types =
           await _fetchDishesUsecase.execute(const NoParams());
+      await _saveDishesUseCase.execute(types); //FIXME its wrong
```
Comment (ibondarovich) [inline features/dishes_menu/lib/src/bloc/dishes_menu_screen/dishes_menu_bloc.dart:38]: ?

File: features/dishes_menu/lib/src/ui/dish_item.dart [hunk 6]
```diff
@@ -26,7 +26,7 @@ class _DishItemState extends State<DishItem> {
   @override
   Widget build(BuildContext context) {
     return GestureDetector(
-      onTap: () => context.navigateTo(
+      onTap: () => context.navigateTo( //FIXME maybe remove in bloc
```
Comment (ibondarovich) [inline features/dishes_menu/lib/src/ui/dish_item.dart:29]: ?

File: features/settings/lib/src/ui/settings_screen.dart [hunk 7]
```diff
@@ -0,0 +1,155 @@
+import 'package:core/core.dart' show BlocBuilder, BlocProvider;
+import 'package:core_ui/core_ui.dart';
+import 'package:flutter/material.dart';
+import 'package:settings/src/bloc/settings/settings_bloc_bloc.dart';
+
+//FIXME rework layout
```
Comment (ibondarovich) [inline features/settings/lib/src/ui/settings_screen.dart:6]: ?

File: features/settings/pubspec.yaml [hunk 8]
```diff
@@ -16,6 +16,8 @@ dependencies:
     path: ../../core
   domain:
     path: ../../domain
+  navigation:
```
Comment (ibondarovich) [inline features/settings/pubspec.yaml:19]: add all navigation dependencies to core and add export of it in core.dart file

File: features/shopping_cart/lib/src/ui/shopping_cart.dart [hunk 9]
```diff
@@ -92,7 +91,7 @@ class ShoppingCartScreen extends StatelessWidget {
                     ],
                   ),
                   SizedBox(
-                    width: AppDimens.size400,
+                    width: AppDimens.size400, //FIXME make adaptive size
```
Comment (ibondarovich) [inline features/shopping_cart/lib/src/ui/shopping_cart.dart:94]: ?

File: features/sign_up_screen/lib/src/ui/sign_up_screen.dart [hunk 10]
```diff
@@ -0,0 +1,97 @@
+import 'package:core/core.dart' show BlocProvider, BlocBuilder;
+import 'package:core/di/app_di.dart';
+import 'package:core/services/auth_service.dart';
+import 'package:core_ui/core_ui.dart';
+import 'package:domain/domain.dart';
+import 'package:flutter/material.dart';
+import 'package:navigation/navigation.dart';
+import 'package:sign_up_screen/src/bloc/sign_up_bloc.dart';
+
+//FIXME make adaptive layout
```
Comment (ibondarovich) [inline features/sign_up_screen/lib/src/ui/sign_up_screen.dart:10]: ?

**Review by ibondarovich — CHANGES_REQUESTED** [review]

File: features/settings/pubspec.yaml (hunk 8 above)
Comment (dungeon-slave) [reply features/settings/pubspec.yaml:19]: Dont get it

**Pull request merged**